browser_patches/firefox/patches/bootstrap.diff

```diff
@@ -4415,8 +4428,61 @@ index 0000000000000000000000000000000000000000..63754937b3464794227fe894b3d6057f
 +    return {defaultPrevented};
 +  }
 +
++  _startDragSessionIfNeeded() {
++    const sess = dragService.getCurrentSession();
++    if (sess) return;
++    dragService.startDragSessionForTests(
++      Ci.nsIDragService.DRAGDROP_ACTION_MOVE |
++        Ci.nsIDragService.DRAGDROP_ACTION_COPY |
++        Ci.nsIDragService.DRAGDROP_ACTION_LINK
++    );
++  }
++
++  _simulateDragEvent(type, x, y, modifiers) {
++    const window = this._frameTree.mainFrame().domWindow();
++    const element = window.windowUtils.elementFromPoint(x, y, false, false);
++    const event = window.document.createEvent('DragEvent');
++
++    event.initDragEvent(
++      type,
++      true /* bubble */,
++      true /* cancelable */,
++      window,
++      0 /* clickCount */,
++      window.mozInnerScreenX + x,
++      window.mozInnerScreenY + y,
++      x,
++      y,
++      modifiers & 2 /* ctrlkey */,
++      modifiers & 1 /* altKey */,
++      modifiers & 4 /* shiftKey */,
++      modifiers & 8 /* metaKey */,
++      0 /* button */, // firefox always has the button as 0 on drops, regardless of which was pressed
```

Comment:
Shouldn't it be actual button for `dragover` events?

Reply:
no, its always 0 for mousemove/dropover events as well.